src/internal/cluster/zarf.go

```diff
@@ -99,24 +98,129 @@ func (c *Cluster) StripZarfLabelsAndSecretsFromNamespaces() {
 	spinner.Success()
 }
 
+// PackageSecretNeedsWait checks if a package component has a running webhook that needs to be waited on.
+func (c *Cluster) PackageSecretNeedsWait(deployedPackage types.DeployedPackage, component types.ZarfComponent, skipWebhooks bool) (needsWait bool, waitSeconds int, hookName string, err error) {
+
+	// Skip checking webhook status when '--skip-webhooks' flag is provided
+	if skipWebhooks {
+		return false, 0, "", nil
+	}
+
+	// Skip checking webhook status on YOLO packages
+	if deployedPackage.Data.Metadata.YOLO {
+		return false, 0, "", nil
+	}
+
+	// Look for the specified component
+	hookMap, componentExists := deployedPackage.ComponentWebhooks[component.Name]
+	if !componentExists {
```

Comment:
These if statements could be collapsed into one return

Reply:
Updated to combine the skipWebhooks and YOLO if statements into a single return statement